Sources/MongoSwift/BSON/Document.swift

```diff
@@ -27,14 +27,26 @@ public class DocumentStorage {
     }
 }
 
+#if swift(>=4.2)
```

Comment:
nit: add a comment explaining the attribute has to be on the type declaration and that's why we did this funky if/else.

Reply:
Done